k2/csrc/array.h

```diff
@@ -332,12 +332,14 @@ class Array2 {
   /* Could view this as num_cols */
   int32_t Dim1() const { return dim1_; }
 
-  ContextPtr &Context() const { return region_->context; }
+  ContextPtr &Context() { return region_->context; }
+
+  const ContextPtr &Context() const { return region_->context; }
```

Comment:
the returned ContextPtr should not be const, this may cause certain complications.

Reply:
Dan, do you mean here we need to get non-const `ContextPtr` in some algorithms with `const Array`? I know there's no such requirements for now (at least for files that have been compiled), but not sure for other algorithms that have not compiled.

Reply:
yes, non-const.  It's OK for const this to return non-const Context, as it's not a member and not really owned by *this.

Reply:
fixed.